pkg/endpoint/policy.go

```diff
@@ -659,6 +659,36 @@ func (e *Endpoint) updateAndOverrideEndpointOptions(opts option.OptionMap) (opts
 	return
 }
 
+// ComputePolicyEnforcement returns whether policy enforcement needs to be
+// enabled for the specified endpoint based off of the global configuration of
+// policy enforcement.
+//
+// Must be called with endpoint mutex held, and repo mutex held for reading.
+func (e *Endpoint) ComputePolicyEnforcement(repo *policy.Repository) (ingress bool, egress bool) {
```

Comment:
From what I can tell, the endpoint mutex is also only a read-lock, which is worth correcting in the comment.

Somewhat alternatively, this does seem pretty close to being a repo function, something like `repo.ShouldEnforcePolicy(labels, endpointIsInit)` although the `IsInit` could also be shuffled around to avoid the boolean. I guess I'm wondering if this function can work with only `OpLabels` which locks itself and won't need any endpoint locks at all. I'm not sure if it's really worth it, however.

Reply:
Agreed for first comment.

Second comment I disagree with. The policy repository code should not be aware of any endpoint-specific functionality.